packages/legend-query/src/stores/postFilterOperators/QueryBuilderPostFilterOperator_Equal.ts

```diff
@@ -90,8 +123,22 @@ export class QueryBuilderPostFilterOperator_Equal extends QueryBuilderPostFilter
         PRIMITIVE_TYPE.FLOAT,
       ] as string[];
 
+      const datePrimitiveTypes = [
+        PRIMITIVE_TYPE.DATE,
+        PRIMITIVE_TYPE.DATETIME,
+        PRIMITIVE_TYPE.STRICTDATE,
+        PRIMITIVE_TYPE.LATESTDATE,
+      ] as string[];
+
       // When changing the return type for LHS, the RHS value should be adjusted accordingly.
       // Numeric value is handled loosely because execution still works if a float (RHS) is assigned to an Integer property(LHS), etc.
+      if (
+        lhsType.path === PRIMITIVE_TYPE.DATETIME &&
+        type !== undefined &&
```

Comment:
`datePrimitiveTypes.includes(type?.path)` will trigger an error, Type 'undefined' is not assignable to type 'string'.
  
`When LHS is of type DateTime, RHS is handled loosely since the operator could be changed to another pure function.  e.g. is -> isOnDay()`.  comments were added.